### DISCUSSION TITLE
Should we go multiaddr

### DISCUSSION THREAD

**Discussion by bryanchriswhite** [pokt-network/pocket#616] [Architecture]
![image](https://user-images.githubusercontent.com/600733/227502458-b91c68ee-4ee1-4208-b900-da6ee5b08859.png)

![image](https://user-images.githubusercontent.com/600733/227502576-8f112f37-6b46-471e-b9f0-dadd9f33fb88.png)

**Comment by bryanchriswhite** [comment]
### Known unknown:

At the time of writing the `getPeerIP` function is being moved from [`libp2p/network/url_conversion.go`](https://github.com/pokt-network/pocket/blob/ef1eba4c996c47c5ee9815dd64b07e4b9c498e41/libp2p/network/url_conversion.go#L132) to [`p2p/utils/url_conversion.go`](https://github.com/pokt-network/pocket/blob/refactor/consolidate-p2p-modules/p2p/utils/url_conversion.go#L133) (see: #576). It contains a `// CONSIDERATION` which is relevant to this discussion:

```
// CONSIDERATION: using a `/dns<4 or 6>/<hostname>` multiaddr instead of resolving here.
// I attempted using `/dns4/.../tcp/...` and go this error:
// > failed to listen on any addresses: [can only dial TCP over IPv4 or IPv6]
```

**Comment by Olshansk** [comment]
Though this does require more thought and discussion, I'm generally in favor of being `multiaddr` native in V1.
